### DISCUSSION TITLE
Requesting permission to use the valkey icon

### DISCUSSION THREAD

**Discussion by Zombach** [valkey-io/valkey#585] [General]
Good afternoon.
Is it possible to obtain permission and how to do this to use valkey icon
![image](https://github.com/valkey-io/valkey/assets/52016832/f9c558ae-fc5d-48af-9238-e73599630623)
in Aspire.Hosting.Valkey resource

[Add ValKey Resource](https://github.com/dotnet/aspire/pull/4324)
[Aspire](https://github.com/dotnet/aspire)

**Comment by stockholmux** [comment]
hey @Zombach, the project has yet to put together a proper logo usage document. Could you tell me how you plan on using it?

**Comment by Zombach** [comment]
> hey @Zombach, the project has yet to put together a proper logo usage document. Could you tell me how you plan on using it?

We want to attach this logo to the nuget package. 
The attached screenshot shows similar options.
![image](https://github.com/valkey-io/valkey/assets/52016832/0f3b0cf1-2367-4dc7-b0c8-f6a0685ef1fe)

Here is an example md

```md
# Aspire.Hosting.ValKey library

Provides extension methods and resource definitions for a .NET Aspire AppHost to configure Cache for ValKey.

## Install the package

In your AppHost project, install the `Aspire.Hosting.ValKey` library with [NuGet](https://www.nuget.org):

```dotnetcli
dotnet add package Aspire.Hosting.ValKey
```

## Usage example

Then, in the _Program.cs_ file of `AppHost`, register a ValKey server and consume the connection using the following methods:

```csharp
var valKey = builder.AddValKey("cache")

var myService = builder.AddProject<Projects.MyService>()
                       .WithReference(valKey);
```

The `WithReference` method configures a connection in the `MyService` project named `cache`. In the _Program.cs_ file of `MyService`, the redis connection can be consumed using the client library [Aspire.StackExchange.Redis](https://www.nuget.org/packages/Aspire.StackExchange.Redis):

```csharp
builder.AddRedisClient("cache");
```

## Additional documentation

* https://valkey.io
* https://github.com/valkey-io/valkey/blob/unstable/README.md
* https://stackexchange.github.io/StackExchange.Redis/Basics
* https://github.com/dotnet/aspire/tree/main/src/Components/README.md

## Feedback & contributing

https://github.com/dotnet/aspire
```

Do I need to add a line using garnet example
_*Garnet MIT License. Copyright (c) Microsoft Corporation.._
Just yours, of course, and which one?
_*ValKey *** License. Copyright (c) ***.._

**Comment by mitchdenny** [comment]
Hi @stockholmux I'm a maintainer for .NET Aspire, @zomach is a contributor who has kindly picked up the effort to add support for Valkey to .NET Aspire. .NET Aspire already has support for Redis and we think that there is some value in having equivalent support for Valkey.

As @Zombach points out this is all about using the icon on the NuGet package that adds this extensibility to .NET Aspire. We like to use icons from projects where we can get the permission because it helps users of Valkey (in this instance) identify the correct package to install visually.

The icon is used would be baked into the package and locked for a specific version of the package, but if the Valkey project changes their branding we can take PRs (or submit PRs ourselves when we notice) if the branding has changed.

I think what we are looking for here is some kind of approval from a designated maintainer of the project to use the icon in our package.

**Reply by Zombach** [reply]
Thank you very much for the clear and competent explanation

**Reply by stockholmux** [reply]
hey @mitchdenny and @Zombach. I'm going to figure out what's permissible here. From a trajectory standpoint, it seems super reasonable to me, but not entirely my call. The project is in a bit of a premature area with the logo usage at the moment, so I don't want to misdirect you.

**Reply by mitchdenny** [reply]
No problems we'll merge the package without the icon and when it gets resolved we can ship an update. Thanks :)

**Comment by celestehorgan** [comment]
Hi @stockholmux - As Valkey is a Linux Foundation project now I think you can probably assume that following rules [Like those the CNCF has for its artwork for various projects](https://github.com/cncf/artwork) is roughly what you'll end up with. Obviously check with your LF liasons, but I suspect you'll end up somewhere here :)
